SH4RE/SH4RE/Login/Tutorial.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  Tutorial.swift
+//  SH4RE
+//
+//  Created by November on 2023-01-04.
+//
+
+import SwiftUI
+
+struct Tutorial: View {
+    @Environment(\.showLoginScreen) var showLoginScreen
+    @State private var tutorialIndex: Int = 0
+    @State private var title: String = ""
+    @State private var description: String = ""
+    @State private var image = UIImage(named: "Tutorial1")!
+
+    func updateText() {
+        switch tutorialIndex {
+        case 0:
+            title = "Find any kind of household item"
+            description = "Welcome to SH4RE, here you can find several household items available to rent. Everything from power tools, outdoors equipment, technology, and so much more!"
+        case 1:
+            title = "Affordable peer-to-peer rates"
+            description = "The items here are listed by users just like you. If you see an item you like, you can make a request to rent it and pick it up directly from the owner."
+            image = UIImage(named: "Tutorial2")!
+        case 2:
+            title = "Get exclusive access to your community"
+            description = "Press the button below to begin SH4RE-ing with your community. You can continue as just a Guest if you want to only take a quick look at what's availble around you.\n\n\n\n"
+            image = UIImage(named: "Tutorial3")!
+        default:
+                break
+        }
+    }
+
+    var body: some View {
+        VStack {
+            Spacer()
+
+            Image(uiImage: image)
+                .resizable()
+                .frame(maxWidth: screenSize.width * 0.75, maxHeight: screenSize.height * 0.3)
+
+            HStack(spacing: 3) {
+                ForEach(0..<3, id: \.self) { index in
+                    Capsule()
+                        .frame(width: index == tutorialIndex ? 50 : 10, height: 10)
+                        .foregroundColor(index == tutorialIndex ? customColours["primaryDark"]! : customColours["grey"]!)
+                        .overlay(Capsule().stroke(Color.gray, lineWidth: 1))
+                        .padding(.bottom, 8)
+                        .animation(.spring(), value: UUID())
+                }
+            }
+
+            Text(title)
+                .font(.system(size: 20, weight: .semibold))
+                .foregroundColor(customColours["primaryDark"]!)
+                .frame(maxWidth: screenSize.width * 0.5)
+                .multilineTextAlignment(.center)
+                .padding()
+                .transition(AnyTransition.opacity.animation(.easeInOut(duration:0.5)))
```

Comment:
Really great use of transitions and animations for sure. I really noticed it right away, especially on the "get started" bubble. Great work!